### PR TITLE
Blog app - controllers

### PR DESCRIPTION
**Pull request by Booze33**
<h1>Hello!</h1>

<h2>In Today's Milestone I...</h2>

- For each URL you should create:

   - A route.
   - An action in the correct controller.
   - A view file.
- View files should include only placeholders written in plain HTML, e.g:

   - Here is a list of posts for a given user
- Make sure that you added code to handle only the 4 required URLs.

<h2> Happy Review</h2>

### CODE REVIEW

**Review by julie-ify — COMMENTED** [review]
Hi @Booze33,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

* The Presentation link on your reader header is broken. Kindly ensure you have no broken readme links.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index], controller: 'users/posts'
+  end
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  resources :posts, only: [:show]
```
Comment (mckent05) [inline config/routes.rb]: - Good job done so far, but this route is not needed as you already have the route above, please remove this 🙏

File: app/controllers/users/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+  end
+end
```
Comment (mckent05) [inline app/controllers/users/posts_controller.rb]: - Good job done so far, but there is no need for this controller, I suggest you remove this and set up your controllers properly 🙏

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index], controller: 'users/posts'
```
Comment (mckent05) [inline config/routes.rb]: - Good job done so far, please add a `show` route for the post controller, this should be handled by the `show` action in the `post` controller. Please fix this 🙏 
- Also, As I mentioned, the `users/posts` controller is not needed, so please remove the reference to that file 👍 👍

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to post_path(@post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
```
Comment (mckent05) [inline app/controllers/posts_controller.rb:22]: - Good job done so far, according to the project requirements, your `post` controller should have an `index` action that handles requests for the corresponding route. Please fix this 👍 👍

File: app/views/layouts/index.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<h1>Here is a list of all posts for a user</h1>
```
Comment (mckent05) [inline app/views/layouts/index.html.erb]: - According to the project requirements, all controllers should have a corresponding view file for each of the controller actions. I.e For the `post#index` action, the corresponding view should be `app/views/posts/index.html.erb`. Please fix this for all controller actions as required 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Booze33, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍
- Good job setting up your controllers 😄  


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- I suggest you fix the broken presentation link as the previous reviewer suggested 👍 👍 
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Booze33,

## APPROVED 🟢

## Highlights
✅ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**